streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -184,7 +184,11 @@ class StreamingContext private[streaming] (
 
   private[streaming] val waiter = new ContextWaiter
 
-  private[streaming] val progressListener = new StreamingJobProgressListener(this)
+  private[streaming] val progressListener = {
+    val streamingListener = new StreamingJobProgressListener(this)
+    sc.ui.get.setStreamingListener(streamingListener)
```

Comment:
This is probably what's causing the unit test failures. The UI might not exist, which is why it's an `Option[SparkUI]` and not just a plain `SparkUI`.